java/spacewalk-java.spec

```diff
@@ -961,16 +966,22 @@ fi
 %{jardir}/taglibs-standard.jar
 %endif
 
-%dir %{realcobsnippetsdir}
-%config %{realcobsnippetsdir}/default_motd
-%config %{realcobsnippetsdir}/keep_system_id
-%config %{realcobsnippetsdir}/post_reactivation_key
-%config %{realcobsnippetsdir}/post_delete_system
-%config %{realcobsnippetsdir}/redhat_register
-%config %{realcobsnippetsdir}/sles_register
-%config %{realcobsnippetsdir}/sles_register_script
-%config %{realcobsnippetsdir}/sles_no_signature_checks
-%config %{realcobsnippetsdir}/wait_for_networkmanager_script
+# owned by cobbler needs cobbler permissions
+%attr(755,root,root) %dir %{cobprofdir}
+%attr(755,root,root) %dir %{cobdirsnippets}
```

Comment:
If we have this, isn't spacewalk-java owning such directories?

 %{_localstatedir}/lib/cobbler/snippets is part of cobbler.

You can't see it explicity at the spec https://build.opensuse.org/package/view_file/systemsmanagement:Uyuni:Master/cobbler/cobbler.spec?expand=1 (but it's packaged by line 307)

So it gets into the RPM: https://build.opensuse.org/package/binaries/systemsmanagement:Uyuni:Master/cobbler/openSUSE_Leap_15.1

Reply:
It is correct, that now also spacewalk-java own that dir, but this is typical SUSE style packaging.
SUSE check, that every directory is owned by at least one package. And we do it during building the package. There are two ways to solve the problem of base directories which must exist as the new package should own new subdirs.

1. Require and build require the other package which own the base dir
2. Include the base dir also in the new package, there for the base dir is owned by different packages.

SUSE often choose option 2 as the buildservice also check dependencies and circles in the dependency chain are much bigger problems then directories owned by multiple packages.

About the file conflict.

This only happens when a directory is owned by multiple packages and they have different opinions about the owner and permissions of the directory.
This was the case here:

Cobbler : root root 755
Spacewalk-java: tomcat tomcat 755

This PR change spacewalk-java to root root 755 as well.

Reply:
@mcalmer: as always, thanks a lot for the explanation, as I was not aware this could be the problem :-)